Feature/61 limit utility nav

stonybrookweb opened the pull request:
Fixes #61 .

### Changes proposed in this pull request

- Limit the output hierarchy depth to 2 items( parent and 1 below ) 

- Limit the number of items to output for each possible submenu to 3

-

### Required pull requests to merge (usually a pull request on Foundation or a plugin)

-

-

### Test sites

Clone these sites to your sandbox, and review them briefly. Does everything work as expected?

- [ ] **Kitchen Sink Site** http://10up-responsi.cms-devl.bu.edu/kitchensink/
- [ ] **BU Landing Pages** http://10up-responsi.cms-devl.bu.edu/bu-landing-pages/
- [ ] **BU Banners** http://10up-responsi.cms-devl.bu.edu/bu-banners/
- [ ] **Faculty Model Site (color palettes)** http://10up-responsi.cms-devl.bu.edu/facultymodel/
- [ ] **BU Bands (light custom CSS)** http://10up-responsi.cms-devl.bu.edu/bands/
- [ ] **Data Sciences (complex custom CSS)** http://10up-responsi.cms-devl.bu.edu/cds-faculty/
- [ ] **Provost Advising (child theme)** http://10up-responsi.cms-devl.bu.edu/advising/

### Review checklist

- [ ] I have tested my changes in my sandbox on Responsive Framework and at least one child theme.
- [ ] I have tested any new filters or action hooks I have introduced in a child theme to ensure they work correctly.
- [ ] I've reviewed the [contribution guidelines](https://github.com/bu-ist/coding-standards/blob/develop/CONTRIBUTING.md).
- [ ] I've updated `CHANGELOG.MD` with a brief explanation of the changes in this pull request in the unreleased section.
- [ ] My code follows [BU Coding Standards](https://github.com/bu-ist/coding-standards).

ashleykolodziej reviewed (changes requested):
https://user-images.githubusercontent.com/1828613/137348835-3335beb6-fbea-4839-903e-63edde714dee.mp4

Can we make it so that the nav item isn't removed when you finish the drag and drop? Details in the video above.

stonybrookweb commented:
@ashleykolodziej I've updated the behavior to reset the menu position to the top/parent level instead of deleting the item on drop.

Updating on the drop reseting the base position was an easy adjustment. 

I'm unsure if I can prevent an item from being nested at this time. I could constantly update the class as an item is in motion but that may cause a lot of JS memory/processing to be used if the process continuously loops while an item is being moved around.

If you'd like me to explore the prevention option I can queue that up hopefully for sometime next week or the week after.

ashleykolodziej commented:
@stonybrookweb I just tested this out and I think we might need to explore either prevention or a way to send updated data to WordPress - it seems like the class is not enough to do that, so the behavior when you save and update is a little funky. WordPress eventually ends up removing the nav item:

https://user-images.githubusercontent.com/1828613/137370453-48092cbc-0cde-4c84-9e32-e31260db0d26.mp4

Let me know what you think - this seems to be a pretty tricky problem. I really like where the interaction and feedback is going though if we can get this figured out.

stonybrookweb commented:
@ashleykolodziej I've updated the whole methodology to reset nested items to the parent level, no more deletions. This is on the database level if an item gets through and on the JS level to reset after drop. There is a hidden span `.is-submenu` that exists for each menu item that is toggled by the WordPress Menu JS, so I was able to leverage that toggle. I've updated the messaging and provided some ideas for testing. From my testing, this seems to be a good workable solution.

Additionally wordsmithing may be helpful to create more user centric messages.

I look forward to your feedback!

stonybrookweb commented:
@ashleykolodziej checking in to see if you need any additional help or information from me to work on the UI.

cc: @sr4136 @smtierneybu
